Added category I4: Cross-Chain (#12)
ComposableSecurity/SCSVS: 2.0/0x300-Integrations/0x304-I4-Cross-Chain.md

| **I4.5** | Verify that the sizes of the destination addresses used in the cross-chained message are verified on the source chain. | 
| **I4.6** | Verify that the cross-chain transfer to zero address will revert if the token uses OpenZeppelin's ERC20 on destination chain, because it reverts on mints to zero address. | 

### Security Requirements for Integration with LayerZero 

| # | Description |
| --- | --- |
| **I4.LZ.1** | Verify that the message sender (msg.sender) of OFT cross-transfer is allowed to transfer tokens on the owner's behalf. |
| **I4.LZ.2** | Verify that the message sender (msg.sender) of ONFT cross-transfer is approved to transfer NFT token or is its owner. |
| **I4.LZ.3** | Verify that the `_from` address (in `_debitFrom` function) specified in the ONFT cross-transfer is the owner of transfered NFT token. |
| **I4.LZ.4** | Verify that the inbound and outbound block confirmation numbers have been set with care, assuming the possible reorg depths on particular chains. |
| **I4.LZ.5** | Verify that the function calls on the destination chain are try-catched and retryable, unless you are builing a blocking application on purpose. |

> | **I4.LZ.6** | Verify that the UA's owner (ideally multisig) is able to call `forceResumeReceive` on the LZ endpoint to unblock the cross-chain channel and protect from DoS. |

Reviewer: ```suggestion
| **I4.LZ.6** | Verify that the UA's owner (ideally multi-sig) is able to call `forceResumeReceive` on the LZ endpoint to unblock the cross-chain channel and protect from DoS. |
```

> | **I4.LZ.7** | Verify that the LayerZero cross-chain call parameters are not hardcoded and can be passed in the function call. That includes: zroPaymentAddress, adapterParams, refundAddress, useZro. |

Reviewer: ```suggestion
| **I4.LZ.7** | Verify that the LayerZero cross-chain call parameters are not hardcoded and can be passed in the function call. That includes: `zroPaymentAddress`, `adapterParams`, `refundAddress`, `useZro`. |
```

| **I4.LZ.8** | Verify that the User Application implementation is able to call the configuration functions on the LZ endpoint (covered if UA inherits from `LzApp`). |
| **I4.LZ.9** | Verify that the User Application has been configured by the protocol or the protocol excplictly accepts the default configuration. |

> | **I4.LZ.10** | Verify that the protocol uses production-ready examples of code prepared by LayerZero team for specific usages (e.g. OFT, ONFT, LzApp, NonBlockingLzApp). |

Reviewer: ```suggestion
| **I4.LZ.10** | Verify that the protocol uses production-ready examples of code prepared by LayerZero team for specific usages (e.g. `OFT`, `ONFT`, `LzApp`, `NonBlockingLzApp`). |
```
